dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java

```diff
@@ -863,7 +889,11 @@ public GHRateLimit getGhRateLimitQuietly() {
         }
 
         try {
-            tags = repo.getRefs("refs/tags/");
+            // this crazy looking structure is because getRefs can result in a cache miss (on repos without tags) whereas listTags seems to not have this problem
```

Comment:
Could this improve our tests but adversely affect prod, especially on a repo like gatk, where due to pagination, several calls will be made to fetch the tags, twice? It will be tricky to figure out, depending on how well the cache works. 

OTOH, there is another ticket to optimize this. And this is for develop. And you have a comment.

Reply:
Not sure, I did the test from the shared google doc with the github app release test on wdl_loc_opt. It was +- a second when I tried it. Note that both calls are paginated anyway, so it should be a constant (but now cached) increase.